library/core/src/slice/mod.rs

```diff
@@ -3994,6 +3994,76 @@ impl<T> [T] {
     }
 }
 
+#[cfg(not(bootstrap))]
+impl<T, const N: usize> [[T; N]] {
+    /// Takes a `&[[T; N]]`, and flattens it to a `&[T]`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(slice_flatten)]
+    ///
+    /// fn print_ints(slice: &[i32]) {
+    ///     for i in slice {
+    ///         println!("{} ", i);
+    ///     }
+    /// }
+    ///
+    /// fn main() {
+    ///     let array = [[1, 2, 3], [4, 5, 6], [7, 8, 9]];
+    ///
+    ///     let flattened = array.flatten();
+    ///     assert_eq!(flattened, &[1, 2, 3, 4, 5, 6, 7, 8, 9]);
+    ///
+    ///     print_ints(flattened);
```

Comment:
We generally don't put `print`s in doctests, because you have to run them to see what that does, and thus they're not checking anything when we run them as tests either.

Some other ideas of things you could include:
```rust
assert_eq!(
    [[1, 2, 3], [4, 5, 6]].flatten(),
    [[1, 2], [3, 4], [5, 6]].flatten(),
);

let slice_of_empty_arrays: &[[i32; 0]] = &[[], [], [], [], []];
assert!(slice_of_empty_arrays.flatten().is_empty());

let empty_slice_of_arrays: &[[u32; 10]] = &[];
assert!(empty_slice_of_arrays.flatten().is_empty());
```

Reply:
Makes sense.